pyvista/core/filters.py

```diff
@@ -4326,6 +4326,44 @@ def strip(poly_data, join=False, max_length=1000, pass_cell_data=False,
         alg.Update()
         return _get_output(alg)
 
+    def shrink(poly_data, shrink_factor=1.0, inplace=False, progress_bar=False):
```

Comment:
This should actually be a part of `DatasetFilters` as the `vtkShrinkFilter` can be applied to any dataset type (at least those that are supported in PyVista)

alluded to in https://github.com/pyvista/pyvista/pull/913#discussion_r498869844

Reply:
Just saw this as well.  This needs to be moved there.

Also, if the input is a `PolyData`, we should return a Data as well, for example:
```py
pd = pyvista.PolyData(shrunk.points, shrunk.cells)
```
And if `overwrite` is used, it will be necessary to convert back to a PolyData before overwriting.

Reply:
@banesullivan @akaszynski Thanks for your discussion. It looks good to me. I will update code :rocket:

Reply:
> Also, if the input is a PolyData, we should return a [Poly]Data as well

Perhaps just do this to easily capture any data arrays as well:

```py
if isinstance(dataset, vtk.vtkPolyData):
    return output.extract_surface()
```

I do not think it is valid to have an `inplace` argument for overwriting the mesh with this filter though